realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -619,6 +622,59 @@ bool BaseRealSenseNode::get_pitch_cb(realsense2_camera_srvs::srv::CameraPitchReq
     return true;
 }
 
+bool BaseRealSenseNode::calibrate_imu_cb(std_srvs::srv::Trigger::Request::SharedPtr req,
+                                         std_srvs::srv::Trigger::Response::SharedPtr res)
+{
+    (void)req;
+    // Check if GYRO and ACCEL are enabled.
+    if (_enable[GYRO] && _enable[ACCEL])
+    {
+        if (_imu_accel_initiated)
+        {
+            // Reset the acceleration vectors
+            _imu_accel_initiated = false;
+            _imu_accel_x_vector.clear();
+            _imu_accel_y_vector.clear();
+            _imu_accel_z_vector.clear();
+
+            // Wait until acceleration values are initiated
+            while (!_imu_accel_initiated)
+            {
+                usleep(1000);
+            };
+
+            // Publish the chassis transform
+            rclcpp::Time current_time = _node.now();
+            // publishChassisTransform(current_time, false, true);
+            _cam_pitch = getImuPitch();
+            RCLCPP_INFO(_node.get_logger(), "Calibrated pitch angle [deg]: %f", _cam_pitch * 57.2958);
+            std_msgs::msg::Float32 pitch_msg;
+            pitch_msg.data = _cam_pitch;
+            _cam_pitch_publisher->publish(pitch_msg);
+
+            // Fill the response values
+            res->success = true;
+            res->message = std::to_string(_cam_pitch);
+            return true;
+        }
+        else
+        {
+            res->success = false;
+            res->message = "Camera calibration could not take place!";
+            return false;
+        }
+    }
+    else
+    {
+        res->success = false;
+        res->message = "Camera angle was calibrated using ENV VAR.";
```

Comment:
IMO the calibrated angle should be returned as well in the service response @pepisg